topics.rb

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require_relative 'octokit_utils'
+require_relative 'options'
+
+def primary_remote
+  ['upstream', 'origin'].map do |name|
+    get_remote(name)
+  end.compact.first
+end
+
+def get_remote(name)
+  require 'rugged'
+  remote = Rugged::Repository.new('.').remotes[name] rescue nil
+  parts  = remote.url.match(/(\w+\/[\w-]+)(?:\.git)?$/) if remote
+  parts[1] if parts
+end
+
+options = parse_options do |opts, result|
+  opts.on('-f', '--fix-topics', 'Add the missing topics to repo') { result[:fix_topics] = true }
+  opts.on('-d', '--delete-topics', 'Delete unwanted topics from repo') { result[:delete_topics] = true }
+
+  opts.on('--repo [REPO]', 'Pass a repository name, defaults to the current upstream.') do |v|
+    result[:remote] = v || primary_remote
+    raise 'Could not guess primary remote. Try using --remote instead.' unless result[:remote]
+  end
+
+  opts.on('--remote REMOTE', 'Name of a remote to work on.') do |v|
+    result[:remote] = get_remote(v)
+    raise "No url set for remote #{v}" unless result[:remote]
+  end
+end
+
+if options[:remote]
+  parsed = { options[:remote] => { 'github' => options[:remote] } }
+else
+  parsed = load_url(options)
+end
+
+util = OctokitUtils.new(options[:oauth])
+
+wanted_topics = [ 'hacktoberfest' ]
```

Comment:
Should this be configurable?

Reply:
yeah, it should, but so should the labels script and I wanted them to behave in the same way. That felt like a bigger change than fit into a single PR.

Reply:
Fair, I'm okay to merge this with the expectation that it will get those updates in the future.